presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsSplit.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tpcds;
+
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.HostAddress;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+// Right now, splits are just the entire TPCH table
+public class TpcdsSplit
+        implements ConnectorSplit
+{
+    private final TpcdsTableHandle tableHandle;
+    private final int totalParts;
+    private final int partNumber;
+    private final List<HostAddress> addresses;
+
+    @JsonCreator
+    public TpcdsSplit(@JsonProperty("tableHandle") TpcdsTableHandle tableHandle,
+            @JsonProperty("partNumber") int partNumber,
+            @JsonProperty("totalParts") int totalParts,
+            @JsonProperty("addresses") List<HostAddress> addresses)
+    {
+        checkState(partNumber >= 0, "partNumber must be >= 0");
+        checkState(totalParts >= 1, "totalParts must be >= 1");
+        checkState(totalParts > partNumber, "totalParts must be > partNumber");
+
+        this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+        this.partNumber = partNumber;
+        this.totalParts = totalParts;
+        this.addresses = ImmutableList.copyOf(requireNonNull(addresses, "addresses is null"));
```

Comment:
`requireNonNull` is not needed if using `ImmutableList.copyOf`. Won't harm though.